src/git.ts

```diff
@@ -1,63 +1,80 @@
 import * as core from "@actions/core";
-import { exec, getExecOutput } from "@actions/exec";
+import { exec, ExecOptions, getExecOutput } from "@actions/exec";
```

Comment:
I'd stay away from reusing `ExecOptions` here. We want a new set of options compatible with both strategies we are using here. As of now, we'd only support `cwd` - something that is common to both. But I have doubt the github API mode would support everything `ExecOptions` has today

Reply:
Defined a new type for this now, also makes `cwd` required, ensuring that we're passing it everywhere